controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -208,7 +221,11 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 				r.log.Error(fmt.Errorf("storageClassClaim is not compatible with existing StorageClass"),
 					"StorageClassClaim validation failed.")
 				r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimFailed
-				return reconcile.Result{}, nil
+				err = r.update(r.storageClassClaim)
+				if err != nil {
+					r.log.Error(err, "failed to update StorageClassClaim", "storageClassClaim", klog.KRef(r.storageClassClaim.Namespace, r.storageClassClaim.Name))
+				}
+				return reconcile.Result{}, err
```

Comment:
Why are we updating the resource we take as input? If the intent was to update the status then just exist (with or without an error) the reconcile function. we have code that ensures that the status will be updated.

Reply:
removed update call